server/src/main/java/org/elasticsearch/index/search/MultiMatchQueryParser.java

```diff
@@ -184,15 +205,7 @@ protected Query newTermQuery(Term term, float boost) {
 
         @Override
         protected Query newPrefixQuery(Term term) {
```

Comment:
A couple notes:
* Although the `cross_fields` query doesn't allow phrases, the `analyzePhrase` method must be implemented to support multi-word synonyms with `auto_generate_synonyms_phrase_query: true`.
* I think we can remove `analyzeMultiPhrase`. This method supports phrase queries that contain a term with synonyms, which should never occur with `cross_fields`. I opted not to remove it since the logic is tricky and I didn't want to risk a regression when we aim to deprecate `cross_fields` anyways.